server/cp/archive_search.py

```diff
@@ -0,0 +1,449 @@
+            **({"slugline": slugline} if not slugline_exact and slugline else {}),
+            **({"phrase": True} if use_phrase else {}),
+            **({"byline": byline} if byline else {}),
+            **(
+                {"from_date": from_date, "to_date": to_date}
+                if self._parse_date(date)
+                else {}
+            ),
+            **({"from_date": from_date} if self._parse_date(from_date) else {}),
+            **({"to_date": to_date} if self._parse_date(to_date) else {}),
+            **({"from_date": raw_from_date} if self._parse_date(raw_from_date) else {}),
+            **({"to_date": raw_to_date} if self._parse_date(raw_to_date) else {}),
+            **{
+                key: ",".join(params[key] or [])
+                for key in self.SELECT_KEYS
+                if key in params
+            },
+        }
+
+        no_text = not any(
+            [qtext, headline, story_text, slugline_exact, slugline, byline]
+        )
+        no_dates = not any([out.get("from_date"), out.get("to_date")])
+        if no_text and no_dates:
+            today = datetime.now(timezone.utc).date()
+            out["from_date"] = (
+                today - timedelta(days=self.default_recent_days)
+            ).isoformat()
+            out["to_date"] = (today + timedelta(days=1)).isoformat()
+
+        return out
+
+    def _api_search(self, api_params: Dict[str, Any]):
+        last_err: Any = None
+
+        with requests.Session() as session:
+            url = f"{self.api_base}{self.api_path}"
+
+            for attempt in range(1, self.max_retries + 1):
+                try:
+                    req = requests.Request(
+                        "GET", url, headers=self.headers, params=api_params
+                    )
+                    prepared = req.prepare()
+                    logger.info("API call (attempt %d): %s", attempt, prepared.url)
+
+                    resp = session.send(prepared, timeout=self.timeout_seconds)
+
+                    if 200 <= resp.status_code < 300:
+                        data = resp.json() if resp.content else {}
+                        return self._normalize_api_response(data)
+
+                    if resp.status_code in self.SEARCH_RETRY_CODES:
+                        last_err = RuntimeError(
+                            f"Retryable status {resp.status_code}: {resp.text[:200]}"
+                        )
+                        self._handle_search_retry(
+                            f"Retryable API error {resp.status_code}", attempt
+                        )
+                        continue
+
+                    raise RuntimeError(
+                        f"Archive API error {resp.status_code}: {resp.text[:500]}"
+                    )
+                except (requests.Timeout, requests.ConnectionError) as e:
+                    last_err = e
+                    self._handle_search_retry(
```

Comment:
you can handle it via [requests session](https://requests.readthedocs.io/en/latest/user/advanced/#example-automatic-retries)